yarn-project/circuit-types/src/messaging/inbox_leaf.ts

```diff
@@ -1,26 +1,27 @@
-import { Fr } from '@aztec/circuits.js';
+import { Fr, INITIAL_L2_BLOCK_NUM, NUMBER_OF_L1_L2_MESSAGES_PER_ROLLUP } from '@aztec/circuits.js';
 import { toBigIntBE } from '@aztec/foundation/bigint-buffer';
 import { BufferReader, serializeToBuffer } from '@aztec/foundation/serialize';
 
 export class InboxLeaf {
   constructor(
-    /** L2 block number in which the message will be included. */
-    public readonly blockNumber: bigint,
-    /** Index of the leaf in L2 block message subtree. */
+    /** Index of the leaf in the whole tree. */
     public readonly index: bigint,
     /** Leaf in the subtree/message hash. */
     public readonly leaf: Fr,
   ) {}
 
   toBuffer(): Buffer {
-    return serializeToBuffer([this.blockNumber, this.index, this.leaf]);
+    return serializeToBuffer([this.index, this.leaf]);
   }
 
   fromBuffer(buffer: Buffer | BufferReader): InboxLeaf {
     const reader = BufferReader.asReader(buffer);
-    const blockNumber = toBigIntBE(reader.readBytes(32));
     const index = toBigIntBE(reader.readBytes(32));
     const leaf = reader.readObject(Fr);
-    return new InboxLeaf(blockNumber, index, leaf);
+    return new InboxLeaf(index, leaf);
+  }
+
+  static smallestIndexFromL2Block(l2block: bigint): bigint {
```

Comment:
I think this would be better to have as a separate utility function as it does not have much to do with `InboxLeaf` class.